frameworks/autoxgboost/setup.sh

```diff
@@ -2,10 +2,21 @@
 HERE=$(dirname "$0")
 VERSION=${1:-"stable"}
 REPO=${2:-"ja-thomas/autoxgboost"}
-# currently both stable and latest maps to master branch
+
+# Version can be specified as 'stable', 'latest', a full 40-character commit hash or a branch
 if [[ "$VERSION" == "latest" || "$VERSION" == "stable" ]]; then
-    VERSION="master"
+  VERSION="master"
+fi
+
+if ! [[ "$VERSION" =~ ^[a-fA-F0-9]{40}$ ]]; then
```

Comment:
asking for the full 40 chars is unusual given that most git commands accept and print only the first few chars of the hash.
It would be more user-friendly to simply use a specific syntax for hashes: `#def123` interpreted as hash `def123`.
Besides this is what was recommended and implemented for autosklearn: https://github.com/openml/automlbenchmark/pull/309

Reply:
I wasn't sure about shorter hashes since over time they wouldn't necessarily have to be unique.
After thinking some more, should that ever happen, we know we should use the first hash in the commit history that matches it, so it's not really a problem. Will change.

Reply:
Done.